dnf/comps.py

```diff
@@ -428,17 +429,19 @@ def _full_package_set(grp):
 
     @staticmethod
     def _pkgs_of_type(group, pkg_types, exclude):
-        def pkgs_update(pkgs, group):
-            pkgs.update(pkg.name for pkg in group
+        def pkgs_update(pkgs, group, substitutions):
+            pkgs.update('.'.join([pkg.name, substitutions['basearch']])
+                        if pkg.basearchonly else pkg.name for pkg in group
                         if pkg.name not in exclude)
 
+        substitutions = dnf.conf.substitutions.Substitutions()
```

Comment:
This won't work if an api user changes the default values (https://dnf.readthedocs.io/en/latest/api_conf.html#dnf.conf.Conf.substitutions).